scripts/deployments/RunWorkflows.ps1

```diff
@@ -9,87 +9,179 @@ OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+
+  .PARAMETER GitHubRepo
+    The GitHub repo to use for the workflow.
+
+  .PARAMETER GitHubRef
+    The GitHub ref to use for the workflow.
+
+  .PARAMETER LoginInteractiveTenantId
+    If set, prompt for credentials and login to the specified tenant.
 
-# Set the environment name which is used to locate configuration files stored within the /config directory 
-$EnvironmentName = "CanadaESLZ-main"
+  .PARAMETER LoginServicePrincipalJson
+    If set, login using the JSON credentials for the specified service principal.
 
-# Replace the Tenant ID with the GUID for your Azure Active Directory instance.
-# It can be found through https://portal.azure.com/#blade/Microsoft_AAD_IAM/ActiveDirectoryMenuBlade/Overview
-$AzureADTenantId = "343ddfdb-bef5-46d9-99cf-ed67d5948783"
+  .PARAMETER WorkingDirectory
+    The directory to use for the workflow.
 
-# Set the working directory.  It should point the root of this project.
-$WorkingDirectory = Resolve-Path "../.."
+  .EXAMPLE
+    PS> .\RunWorkflows.ps1 -EnvironmentName CanadaESLZ-main -LoginInteractiveTenantId '8188040d-6c67-4c5c-b112-36a304b66dad' -DeployManagementGroups
 
-$Features = @{
-  # Prompt to login to Azure AD and set the context for Azure deployments
-  PromptForLogin = $false
+    Deploy management groups interactively.
 
-  # Resource Organization
-  DeployManagementGroups = $false
+  .EXAMPLE
+    PS> .\RunWorkflows.ps1 -EnvironmentName CanadaESLZ-main -LoginInteractiveTenantId '8188040d-6c67-4c5c-b112-36a304b66dad' -DeployManagementGroups -DeployRoles -DeployLogging -DeployPolicies -DeployHubNetworkWithAzureFirewall
 
-  # Access Control
-  DeployRoles = $false
+    Deploy all platform components interactively, with Azure Firewall.
 
-  # Logging
-  DeployLogging = $false
+  .EXAMPLE
+    PS> .\RunWorkflows.ps1 -EnvironmentName CanadaESLZ-main -LoginInteractiveTenantId '8188040d-6c67-4c5c-b112-36a304b66dad' -DeploySubscriptions 'a188040e-6c67-4c5c-b112-36a304b66dad,7188030d-6c67-4c5c-b112-36a304b66dac'
 
-  # Guardrail & Compliance
-  DeployPolicy = $false
+    Deploy 2 subscriptions interactively.
 
-  # Hub Networking - With Network Virtual Appliance
-  DeployHubNetworkWithNVA = $false
+  .EXAMPLE
+    PS> .\RunWorkflows.ps1 -GitHubRepo 'Azure/CanadaPubSecALZ' -GitHubRef 'refs/head/main' -LoginServicePrincipalJson '{ <output from: az ad sp create-for-rbac> }' -DeployManagementGroups
 
-  # Hub Networking - With Azure Firewall
-  DeployHubNetworkWithAzureFirewall = $false
+    Deploy management groups using service principal authentication.
 
-  # Subscriptions
-  DeploySubscriptions = $false
+    The action in the GitHub workflow could look like this:
+
+    - name: Deploy Management Groups
+      run: |
+        ./RunWorkflows.ps1 `
+          -DeployManagementGroups `
+          -LoginServicePrincipalJson '${{secrets.AZURE_CREDENTIALS}}' `
+          -GitHubRepo ${env:GITHUB_REPOSITORY} `
+          -GitHubRef ${env:GITHUB_REF}
+#>
+
+[CmdletBinding()]
+Param(
```

Comment:
Let's update the input to take in NVA Username and Password as input.  It would also be required when deploying Hub + NVA scenario.

Reply:
done